charts/terra-argocd-app/templates/app.yaml

```diff
@@ -1,39 +1,45 @@
-{{- with .Values -}}
+{{- if .Values.releaseType | eq "app" }}
+# Generates an ArgoCD application for a Terra application release
+{{- $app := required "releaseName is required" .Values.releaseName }}
```

Comment:
Is there a reason we can't use `Release.Name` [link](https://helm.sh/docs/chart_template_guide/builtin_objects/)?

Reply:
`Release.Name` can't be used here. The `releaseName` value in this context refers to the name of the service that is being deployed by this ArgoCD application (eg. `cromwell`, `workspacemanager`, `diskmanager`). `Release.Name` would be the release name for this Helm chart ([eg. `terra-argocd-app-cromwell`](https://github.com/broadinstitute/terra-helmfile/blob/master/helmfile.yaml#L149))

Reply:
Oh I see! That makes sense.